server/src/main/java/org/elasticsearch/plugins/Plugin.java

```diff
@@ -192,6 +193,17 @@ public void onIndexModule(IndexModule indexModule) {
         return UnaryOperator.identity();
     }
 
+    /**
+     * Provides the list of this plugin's custom thread pools, empty if
+     * none.
+     *
+     * @param settings the current settings
+     * @return executors builders for this plugin's custom thread pools
+     */
+    public List<ExecutorBuilder<?>> getExecutorBuilders(Settings settings) {
```

Comment:
We've just recently removed that plugin interface method, not sure if we really want to re-introduce.
Maybe we could tackle the todo here instead? https://github.com/crate/crate/pull/10024/files#diff-dcad7c2b3e3caa630e82bfbbcd7a6e2cR122

Reply:
I had a look to port this over to the new Azure SDK 11, unfortunately there seems to be a problem with the SAS Token authentication related to bulk deletes and elasticsearch reverted them: 
https://github.com/elastic/elasticsearch/commit/c94be58b5a2116d3a8462dbd6d1ba72358768649
My understanding is that the standard way of authentication on the Azure Blob Storage are SAS Token and they don't work with the bulk deletes. Why did we remove the `getExecutorBuilders`  method initial on the plugin interface ?

Reply:
Ok I see. Maybe we've just removed this plugin methods as they were unused. @mfussenegger any input here?

Reply:
>  Maybe we've just removed this plugin methods as they were unused

Yes, that was one of the reasons for the removal, but also because I think we should be careful with adding additional threads, as they generally add overhead and have performance impact. E..g spawning threads for azure even if users don't use snapshots on azure would be a waste.

I'd rather we followed up on looking into the latest SDK if the SAS token issue is fixed before re-introducing the additional threadpool